engine/src/sol/retry_rpc.rs

```diff
@@ -208,9 +209,28 @@ impl SolRetryRpcApi for SolRetryRpcClient {
 				Box::pin(move |client| {
 					let encoded_transaction = encoded_transaction.clone();
 					#[allow(clippy::redundant_async_block)]
-					Box::pin(
-						async move { client.send_transaction(encoded_transaction, config).await },
-					)
+					Box::pin(async move {
+						let tx_signature =
+							client.send_transaction(encoded_transaction, config).await?;
+
+						for status_retry in 0..GET_STATUS_BROADCAST_RETRIES {
+							let signature_statuses =
+								client.get_signature_statuses(&[tx_signature], true).await?;
+							match signature_statuses.value.first() {
+								Some(_) => return Ok(tx_signature),
+								None => {
+									// Ignore sleep at last step
+									if status_retry < GET_STATUS_BROADCAST_RETRIES {
```

Comment:
This is never false.

Reply:
I'd suggest using a interval/make_periodic_tick and putting the sleep before instead of after, that way you avoid the need to count anything.

Reply:
I implemented it so it had the same behavior as the rust client but makes sense what you said. Just pushed an update using that.